docs/configuration.asciidoc

```diff
@@ -15,20 +13,319 @@ server_url: 'http://localhost:8200'
+
+[float]
+[[config-source-lines-error-app-frames]]
+==== `source_lines_error_app_frames`
+[float]
+[[config-source-lines-error-library-frames]]
+==== `source_lines_error_library_frames`
+[float]
+[[config-source-lines-span-app-frames]]
+==== `source_lines_span_app_frames`
+[float]
+[[config-source-lines-span-library-frames]]
+==== `source_lines_span_library_frames`
+
+|============
+| Environment                                     | `Config` key                        | Default
+| `ELASTIC_APM_SOURCE_LINES_ERROR_APP_FRAMES`     | `source_lines_error_app_frames`     | `5`
+| `ELASTIC_APM_SOURCE_LINES_ERROR_LIBRARY_FRAMES` | `source_lines_error_library_frames` | `5`
+| `ELASTIC_APM_SOURCE_LINES_SPAN_APP_FRAMES`      | `source_lines_span_app_frames`      | `0`
+| `ELASTIC_APM_SOURCE_LINES_SPAN_LIBRARY_FRAMES`  | `source_lines_span_library_frames`  | `0`
+|============
+
+By default, the APM agent collects source code snippets for errors.
+With the above settings, you can modify how many lines of source code is collected.
+
+We differ between errors and spans, as well as library frames and app frames.
+
+WARNING: Especially for spans, collecting source code can have a large impact on
+storage use in your Elasticsearch cluster.
+
+[float]
+[[config-max-queue-size]]
+==== `max_queue_size`
+
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_MAX_QUEUE_SIZE` | `max_queue_size` | `500`
+|============
+
+Maximum queue length of transactions before sending transactions to the APM server.
+A lower value will increase the load on your APM server,
+while a higher value can increase the memory pressure of your app.
+A higher value also impacts the time until transactions are indexed and searchable
+in Elasticsearch.
+
+This setting is useful to limit memory consumption if you experience a sudden spike
+of traffic.
+
+[float]
+[[config-flush-interval]]
+==== `flush_interval`
+
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_FLUSH_INTERVAL` | `flush_interval` | `10`
+|============
+
+Interval with which transactions should be sent to the APM server, in seconds.
+A lower value will increase the load on your APM server,
+while a higher value can increase the memory pressure on your app.
+A higher value also impacts the time until transactions are indexed and searchable
+in Elasticsearch.
+
+[float]
+[[config-transaction-sample-rate]]
+==== `transaction_sample_rate`
+
+|============
+| Environment                           | `Config` key              | Default
+| `ELASTIC_APM_TRANSACTION_SAMPLE_RATE` | `transaction_sample_rate` | `1.0`
+|============
+
+By default, the agent will sample every transaction (e.g. request to your service).
+To reduce overhead and storage requirements, you can set the sample rate to a value
+between `0.0` and `1.0`.
+We still record overall time and the result for unsampled transactions, but no
+context information, tags, or spans.
```

Comment:
Did you align with the other agent owners on the default value? I am surprised that you start with 1.0.